// movie-list-node/index.js
// This product uses the TMDB API but is not endorsed or certified by TMDB
function infoConsole(message) {
    return console.log(colors.black(message).bgBrightCyan);
}
function successConsole(message) {
    return console.log(colors.red(message).bgYellow);
}
const API_KEY = "01b0b94679a44fe7a5b9646f0056cba8";

module.exports = async function getMovieList(movieOrSeriesName){
    const axios = require("axios");
    const fs = require('fs');
    const json2xls = require('json2xls');
    infoConsole("Creating Axios call to Movie List");
    var dirOutput =  "./files/output/";
    var options = {
        method: 'GET',
        url: 'https://api.themoviedb.org/3/search/movie?api_key='+API_KEY+'&query='+movieOrSeriesName
      };

    await axios.request(options).then(function (response) {
        //var responseDataArr = [response.data.results[0].title, response.data.results[0].vote_average, response.data.results[0].overview, response.data.results[0].release_date];
        //var responseDataArr = [response.data.results.title, response.data.results.vote_average, response.data.results.overview];
        //console.log(response.data.results);
        var responseDataArr = response.data.results;
        infoConsole("Creating Excel File");
        var xls = json2xls(responseDataArr,{
            fields: {title:{ caption: "title", type:"string"}, release_date:{ caption: "release_date", type:"string"}, popularity:{ caption: "popularity", type:"string"}}
        });
        fs.writeFileSync(dirOutput + "movie-list.xlsx", xls, "binary");
        successConsole("Successfully created excel file for Movie List");
        return responseDataArr;
    }).catch(function (error) {
        console.error(error);
    });
};

// module.exports = async function getMovieList(movieOrSeriesName){
//     const axios = require("axios");
//     var options = {
//         method: 'GET',
//         url: 'https://api.themoviedb.org/3/search/movie?api_key='+API_KEY+'&query='+movieOrSeriesName
//       };

//     await axios.request(options).then(function (response) {
//         //var responseDataArr = [response.data.results[0].title, response.data.results[0].vote_average, response.data.results[0].overview, response.data.results[0].release_date];
//         //var responseDataArr = [response.data.results.title, response.data.results.vote_average, response.data.results.overview];
//         //console.log(response.data.results);
//         var responseDataArr = response.data.results;
//         return responseDataArr;
//     }).catch(function (error) {
//         console.error(error);
//     });
// };
